codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -546,6 +581,7 @@ void incrementStreamableBytesForTree(int numBytes) {
          */
         private final class Frame {
             final ByteBuf data;
+            final Http2Headers headers;
```

Comment:
I think it may be re-factored at that time, but I think this change justifies an attempt.

What about a simple interface:

```java
interface FramePayload {
  int size();
  void release();
  /**
   * @return {@code true} if a flush is needed. {@code false} otherwise.
   */
  boolean write();
}
```

then `Frame` would have a member of type `FramePayload` instead of `ByteBuf` or `Http2Headers`.